src/steadystateproblem.cpp

```diff
@@ -238,45 +243,71 @@ void SteadystateProblem::computeSteadyStateQuadrature(NewtonSolver *newtonSolver
      We therefore compute the integral over xB first and then do a
      matrix-vector multiplication */
 
-    /* Compute the integral over the adjoint state xB:
-       If the Jacobian has full rank, this has an anlytical solution, since
-           d/dt[ xB(t) ] = JB^T(x(t), p) xB(t) = JB^T(x_ss, p) xB(t)
-       This linear ODE system with time-constant matrix has the solution
-           xB(t) = exp( t * JB^T(x_ss, p) ) * xB(0)
-       This integral xBI is given as the solution of
-           JB^T(x_ss, p) * xBI = xB(0)
-       So we first try to solve the linear system, if possible. */
+    /* Try to compute the analytical solution for quadrature algebraically */
+    getQuadratureByLinSolve(newtonSolver, model);
+
+    /* Analytical solution didn't work, perform simulation instead */
+    if (!hasQuadrature())
+        getQuadratureBySimulation(solver, model);
```

Comment:
No, we just call that once:
`workSteadyStateBackwardProblem()` calls `computeSteadyStateQuadrature()` calls `getQuadratureBySimulation()` calls `getSteadystateSimulation()`, and only the last one actually carries out integration. So there's only one.
Admittedly, all these functions are rather short and one could avoid maybe one level of nesting, but I preferred having it this way...

Reply:
You are right, the forward part is only called during `workSteadyStateBackwardProblem` and the backward part is `workSteadyStateProblem`, so thats fine. 

I think the nesting is absolutely fine, what I am more concerned about is the dual use of `getSteadystateSimulation`. Depending on whether `backward` is false or not, those routines do completely different things, even though they are using the same code. I think it would be cleaner to split the functionality in three functions: two functions that implement everything that is common specific to `backward=true` and `backward=false` and one function that handles the commonality between the two, i.e. the simulation + convergence checks.

Reply:
Ignore the previous comment, I think it doesnt actually fix much as there is no good way to work around the `solver->writeSolution` calls. I think the best thing to do here is add a comment in the code that there are two ways to end up in this function and that they serve two different purposes.